core/state_transition.go

```diff
@@ -76,7 +76,7 @@ type Message interface {
 }
 
 // IntrinsicGas computes the 'intrinsic gas' for a message with the given data.
-func IntrinsicGas(data []byte, contractCreation, homestead bool) (uint64, error) {
+func IntrinsicGas(data []byte, contractCreation, homestead bool, istanbul bool) (uint64, error) {
```

Comment:
It's really nasty to pass a batch of fork indicators. What about using a `dynasty`?

Reply:
We could pass the ChainConfig perhaps?

Reply:
or the `chainRules`

Reply:
Yes, but we use this function in some tests, so it seems weird to construct the `chainRule`, maybe worse than `homestead, istanbul`. So what's your prefer? `ChainRules` can make normal path code nicer, but test is a bit dirty

Reply:
Ok, I agree we can have flags. But actually, I'd personally prefer to have not `istanbull bool` but instead `eip2028Active bool` (and possibly `eip155Active bool`).